content/06.know-your-problem.md

```diff
@@ -1 +1,18 @@
 ## Rule 4: Know your data and your question
+
+In the era of easily accessible datasets, one sometimes starts analyzing data without a good understanding of the study design, namely why the data were collected and how. 
+Having appropriate meta-data, a comprehensive data dictionary, and even the actual data collection protocol is essential for any analysis, including one that involves deep learning. 
+It's even better to also have access to a subject matter expert who has either collected or analyzed this type of data before! 
+For example, if the main reason why the data were collected was to test the impact of an intervention, it may be the case that a randomized controlled trial was performed.
+However, ethical or other study considerations may make this impossible or impractical, in which case the design may have been an observational one, either prospective or retrospective.
+These designs may also incorporate some amount of matching - for example, cases and controls may be selected so that the age range or weight distribution is similar.
+All of these different designs have different assumptions and caveats, which cannot be ignored during a data analysis.
+Many datasets are now passively collected or do not have a specific design, but even in this case it is important to know how individuals or samples were treated (for example, if all samples are from the same study site, if certain ethnic groups or zip codes are oversampled, if there are differences in processing dates or techniques.)
+Systematic biases can lead to artifacts or "batch effects," which mean that instead of finding correlates with an outcome or grouping of interest, the investigator may find correlates with variables that are not of interest and obtain misleading results [@doi:10.1038/nrg2825].
+Other study design considerations that should not be overlooked include knowing whether a study involves biological or technical replicates or both.
+For example, are some samples collected from the same individuals at different time points? Are those time points before and after some treatment?
+If one assumes that all the samples are independent but that is in fact not the case, a variety of issues may arise, including having a lower effective sample size than expected.
+In general, deep learning has an increased tendency for overfitting, compared to classical methods, due to the large number of parameters being estimated, making issues of adequate sample size even more important (see [Rule 7](https://github.com/Benjamin-Lee/deep-rules/blob/master/content/09.overfitting.md)).
```

Comment:
@dhimmel any ideas on how to do automatic numbering here?  Each rule is currently in it's own md file, but we would want to link to the manuscript section, not the md file.  [Section labels](http://lierdakil.github.io/pandoc-crossref/#section-labels) may be close to what we want.

Reply:
I think you can create the links with anchors.

You could go:

```md
[Rule 7](#rule-7-address-deep-neural-networks-increased-tendency-to-overfit-the-dataset)
```

Or define a nicer ID for Rule 7 header:

```
## This is the header at the top level of rule 7 {#overfitting-rule}
```

Then in this document you can do:

```md
[Rule 7](#overfitting-rule)
```

But do you want the rule numbering to be automated as well?

Reply:
Thanks @dhimmel, `[Rule 7](#overfitting-rule)` with the corresponding updated header looks like the right solution for now.

If we want to automate rule numbering, we might need to customize the sections more with pandoc-crossref or turn to jinja2.  That could be a separate issue or pull request.